CDCgov/prime-simplereport
# Johanna/3661 makes test results heading sticky

# FRONTEND PULL REQUEST

## Related Issue

Fixes [#3661](https://github.com/CDCgov/prime-simplereport/issues/3661)

## Changes Proposed

As part of the acceptance criteria. Three elements in the test results page will have a top sticky behavior (The results title, the results filter and the table header). To simplify css styling the first change was to group these elements into a div container that leverages the new class "sticky-heading". The grouping makes it really easy to ensure that these three components will always naturally stack one on top of the other no matter the resolution of the page. 

The other important change introduced was the addition of a "fake" table heading. Again this decision boiled down to making sure that the heading elements would always naturally stack one after the other. This "fake" heading will be ignore by screen readers. The results table will continue to have the regular table heading in order to not affect the page semantics but it will be hidden from display moving forward.

A different approach could have been not creating a fake table heading, applying a sticky position to the regular table heading and setting the position not to "0" but to whatever the height of the other heading elements combined is. The issues with this other option arise because the other elements do not have static sizes but a responsive behavior. Which not only makes the required height to be available only after page load but also prone to changes if the user decides to resize the browser window. This second option would required javascript coding to catch different browser events, on the fly calculations and modifications to the css (which sometimes can feel sloppy imo).


## Additional Information
As part of this change I also adjusted the width percentages of the different results columns. It seemed that the ones setup previously were being overwritten by the browser because they were not accommodating well the contents of certain columns like "action". When the multiplex changes are introduced and new columns get added to this table these percentages will need to be revisited again. 

## Testing

1) Load a facility that has 20 or more test results registered.
2) Go to "Results" view and scroll all the way to the end of the page.
3) The heading area (Title area, filter area and the header of the results table) should have a sticky behavior.
 
## Screenshots / Demos
**Initial state**
<img width="1495" alt="Screen Shot 2022-05-05 at 12 25 44 PM" src="https://user-images.githubusercontent.com/103958711/166969048-376f4efe-8af8-4561-8b3c-dc4696c95911.png">

**Sticky behavior**
<img width="1499" alt="Screen Shot 2022-05-05 at 12 25 53 PM" src="https://user-images.githubusercontent.com/103958711/166969061-15637563-f8e5-44ce-b2d9-19314deab624.png">


## Checklist for Author and Reviewer

### Design

### Content
- [n/a] Any content changes have been approved by content team

### Support
- [n/a] Any changes that might generate new support requests have been flagged to the support team
- [n/a] Any changes to support infrastructure have been demo'd to support team

### Security
- [n/a] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [n/a] Any dependencies introduced have been vetted and discussed
